app/code/Magento/Cms/Controller/Router.php

```diff
@@ -112,6 +112,12 @@ public function match(\Magento\Framework\App\RequestInterface $request)
         $page = $this->_pageFactory->create();
         $pageId = $page->checkIdentifier($identifier, $this->_storeManager->getStore()->getId());
         if (!$pageId) {
+            $pageId = $page->checkIdentifier($identifier, $this->_storeManager->getDefaultStoreView()->getId());
+            if($pageId) {
+                $this->_response->setRedirect('/');
```

Comment:
Bad practice to hardcode home Url, better way to use \Magento\Store\Model\Store::getHomeUrl() method